### PR TITLE
`RollCall` changes

### PR DESCRIPTION
**Pull request by rob-maron**
Closes https://github.com/EspressoSystems/HotShot/issues/3455

## This PR:
Adds a message handling framework to handle `External` HotShot events. These are messages that are "external" to HotShot but that HotShot subscribers may want to about. It's like an out-of-band communication layer and prevents us from having to use a separate network to disseminate the information.

Adds `RollCall` functionality. The frontend will use this to aggregate node identities, starting with the public API URL. On startup a node's `RollCallInfo` will be broadcast the network. Past that, we will have an external service occasionally sending these messages (e.g. when it restarts) and providing the data to the frontend.

Adds a Public API URL as an optional argument. If not supplied, we will not send a `RollCall` message.

### CODE REVIEW

**Comment by rob-maron** [comment]
We need to wait for HotShot to be updated before this gets merged

**Review by imabdulbasit — APPROVED** [review]
LGTM! just a minor comment.

File: sequencer/src/external_event_handler.rs
Comment (imabdulbasit) [inline sequencer/src/external_event_handler.rs:113]: we can use bail!() here

File: sequencer/src/external_event_handler.rs
Comment (imabdulbasit) [inline sequencer/src/external_event_handler.rs:57]: I think the event handling will be quick, but should the channel capacity be increased just to be on the safe side?

**Pull request closed without merging**
